src/test/java/picard/sam/markduplicates/AbstractMarkDuplicatesCommandLineProgramTester.java

```diff
@@ -163,15 +164,20 @@ public void test() throws IOException {
                 for (final SAMRecord record : reader) {
                     outputRecords++;
                     final String key = samRecordToDuplicatesFlagsKey(record);
-
+                    if (!this.duplicateFlags.containsKey(key)) {
+                        System.err.println("DOES NOT CONTAIN KEY: " + key);
+                    }
                     Assert.assertTrue(this.duplicateFlags.containsKey(key));
                     final boolean value = this.duplicateFlags.get(key);
                     this.duplicateFlags.remove(key);
                     Assert.assertEquals(record.getDuplicateReadFlag(), value, "Mismatching read: " + record.getSAMString());
                     if (testOpticalDuplicateDTTag && MarkDuplicates.DUPLICATE_TYPE_SEQUENCING.equals(record.getAttribute("DT"))) {
                         sequencingDTErrorsSeen.add(record.getReadName());
                     }
+                    Assert.assertEquals(record.getDuplicateReadFlag(), value);
                 }
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Do you really want the test to pass if an exception is thrown here? It seems like you'd want is to rethrow using an unchecked exception, so the test fails.
```suggestion
                throw new RuntimeException(e);
```